### PR TITLE
Add Manufacturer Data Filter Web Bluetooth sample

### PR DESCRIPTION
**Pull request by beaufortfrancois**
This sample adds a Web Bluetooth sample for the upcoming Manufacturer Data filter. See https://github.com/WebBluetoothCG/web-bluetooth/pull/545

Live preview: 
- https://beaufortfrancois.github.io/samples/web-bluetooth/manufacturer-data-filter-async-await.html
- https://beaufortfrancois.github.io/samples/web-bluetooth/manufacturer-data-filter.html
- https://beaufortfrancois.github.io/samples/web-bluetooth/index.html

@reillyeon Can you review?

image

### CODE REVIEW

File: web-bluetooth/manufacturer-data-filter-async-await.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+async function onButtonClick() {
+  let manufacturerData = {};
```
Comment (reillyeon) [inline web-bluetooth/manufacturer-data-filter-async-await.js:2]: nit: Const, here and for most other variables in the example.

File: web-bluetooth/manufacturer-data-filter.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+function onButtonClick() {
+  let manufacturerData = {};
```
Comment (reillyeon) [inline web-bluetooth/manufacturer-data-filter.js:2]: Same here.

**Review by reillyeon — APPROVED** [review]

File: web-bluetooth/manufacturer-data-filter-async-await.js (hunk 1 above)
Comment (beaufortfrancois) [reply web-bluetooth/manufacturer-data-filter-async-await.js:2]: I've replaced `let` with `const` when appropriate in other parts of this example. 
This one was not replaced as I wanted to convey this variable would be updated later.

File: web-bluetooth/manufacturer-data-filter.js (hunk 2 above)
Comment (beaufortfrancois) [reply web-bluetooth/manufacturer-data-filter.js:2]: Ditto.

**Review by ttloves2 — COMMENTED** [review]
AlphaSTAR

**Pull request merged**